File: 3779.minimum-number-of-operations-to-have-distinct-elements.java
#
# @lc app=leetcode id=3779 lang=java
#
# [3779] Minimum Number of Operations to Have Distinct Elements
#
# @lc code=start
class Solution {
    public int minOperations(int[] nums) {
        if (nums.length <= 1) return 0;
        int operations = 0;
        Set<Integer> uniqueElements = new HashSet<>();
        while (true) {
            for (int i = 0; i < nums.length; i++) {
                uniqueElements.add(nums[i]);
            }
            if (uniqueElements.size() == nums.length) break; // All elements are distinct.
            operations++; // Count this operation.
            if (nums.length <= 3) break; // If fewer than three elements remain, operation ends array.
            nums = Arrays.copyOfRange(nums, 3, nums.length); // Remove first three elements.
            uniqueElements.clear(); // Reset unique tracking for new subarray.
        }
        return operations;
    }
}
# @lc code=end

File: 3779.minimum-number-of-operations-to-have-distinct-elements.go
# @lc app=leetcode id=3779 lang=golang
#
# [3779] Minimum Number of Operations to Have Distinct Elements
#
# @lc code=start
func minOperations(nums []int) int {
    operations := 0
    
    for containsDuplicates(nums) {
        if len(nums) <= 3 {
            // Remove all remaining elements if fewer than three
            nums = []int{}
        } else {
            // Remove the first three elements
            nums = nums[3:]
        }
        operations++
    }
    
    return operations
}

func containsDuplicates(nums []int) bool {
    elementCount := make(map[int]bool)
    for _, num := range nums {
        if elementCount[num] {
            return true
        }
        elementCount[num] = true
    }
    return false
}
# @lc code=end